v19.1/gssapi_authentication.md

```diff
@@ -0,0 +1,106 @@
+
+## Configuring the CockroachDB node
+1. Copy the keytab file at a location accessible by the cockroach binary.
+
+2. [Create certificates](create-security-certificates.html) for internode and root user authentication:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ mkdir certs my-safe-directory
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ cockroach cert create-ca \
+    --certs-dir=certs \
+    --ca-key=my-safe-directory/ca.key
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ cockroach cert create-node \
+    localhost \
+    $(hostname) \
+    --certs-dir=certs \
+    --ca-key=my-safe-directory/ca.key
+    ~~~
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ cockroach cert create-client \
+    root \
+    --certs-dir=certs \
+    --ca-key=my-safe-directory/ca.key
+    ~~~
+
+3. Provide the path to the keytab in the `KRB5_KTNAME` environment variable:
+
+      {% include copy-clipboard.html %}
+      ~~~ shell
+      $ KRB5_KTNAME=<path to keytab file>
+      ~~~
+
+4. Start a CockroachDB node:
+
+      {% include copy-clipboard.html %}
+      ~~~ shell
+      $ cockroach start \
+      --certs-dir=certs \
+      --listen-addr=0.0.0.0
+      ~~~
+
+5. Connect to CockroachDB as root using the root client certificate generated above:
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    $ cockroach sql --certs-dir=certs
+    ~~~
+
+6. [Enable an enterprise license](enterprise-licensing.html#obtain-a-trial-or-enterprise-license-key).
+
+7. Enable GSSAPI authentication:
+
+    {% include copy-clipboard.html %}
+    ~~~ shell
+    > SET cluster setting server.host_based_authentication.configuration = 'host all all all gss include_realm=0';
+    ~~~
+
+  Setting the `server.host_based_authentication.configuration` cluster setting makes it mandatory for all users (except `root`) to authenticate using GSSAPI. The `root` user is still required to authenticate using its client certificate.
+
+  The `include_realm=0` option is required to tell CockroachDB to remove the `@DOMAIN.COM` realm information from the username. We don't support any advanced mapping of GSSAPI usernames to CockroachDB usernames right now. If you want to limit which realms' users can connect, you can also add one or more `krb_realm` parameters to the end of the line as a whitelist, as follows: `host all all all gss include_realm=0 krb_realm=domain.com krb_realm=corp.domain.com`
+
+## Configuring the client
+
+{{site.data.alerts.callout_info}}
+The cockroach sql shell does not yet support GSSAPI authentication. You need to use a GSSAPI-compatible Postgres client, such as Postgres's `psql` client.
+{{site.data.alerts.end}}
+
+1. Install and configure your Kerberos client.
+2. Install the Postgres client (for example, postgresql-client-10 Debian package from postgresql.org).
+3. Get a Kerberos TGT from Active Directory using `kinit`.
```

Comment:
Should we remove "from Active Directory" here since they might not be using AD?

Reply:
good catch, yes, "from the KDC" would be better.